lib/iris/mesh/components.py

```diff
@@ -2999,22 +3006,30 @@ def __getitem__(self, keys):
         return self.copy()
 
     def update_from_mesh(self):
-        try:
-            object.__setattr__(self, "_updating", True)
-            if (self._last_modified is None) or (
-                self._last_modified < self.mesh._last_modified
-            ):
+        """Fetch and recalculate the points and bounds from the relevant coord on the mesh.
+
+        This uses _last_modified to ensure that points/bounds are only updated when they
+        are out of date with the mesh, to prevent unnecessary time costs.
+
```

Comment:
This bit isn't true any more : this routine now updates regardless, and the timestamp check is done in the getattribute override.